plugins/techdocs-react/api-report.md

```diff
@@ -0,0 +1,116 @@
+## API Report File for "@backstage/plugin-techdocs-react"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { AsyncState } from 'react-use/lib/useAsync';
+import { ComponentType } from 'react';
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import { Context } from 'react';
+import { Dispatch } from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { Extension } from '@backstage/core-plugin-api';
+import { default as React_2 } from 'react';
+import { SetStateAction } from 'react';
+import { VersionedValue } from '@backstage/version-bridge';
+
+// @alpha
+export function createTechDocsAddon<TComponentProps>(
+  options: TechDocsAddonOptions<TComponentProps>,
+): Extension<ComponentType<TComponentProps>>;
+
+// @alpha (undocumented)
+export const defaultTechDocsReaderPageValue: TechDocsReaderPageValue;
+
+// @alpha
+export const TECHDOCS_ADDONS_WRAPPER_KEY = 'techdocs.addons.wrapper.v1';
+
+// @alpha
+export enum TechDocsAddonLocations {
+  CONTENT = 'content',
+  HEADER = 'header',
+  PRIMARY_SIDEBAR = 'primary sidebar',
+  SECONDARY_SIDEBAR = 'secondary sidebar',
+  SUBHEADER = 'subheader',
+}
+
+// @alpha
+export type TechDocsAddonOptions<TAddonProps = {}> = {
+  name: string;
+  location: TechDocsAddonLocations;
+  component: ComponentType<TAddonProps>;
+};
+
+// @alpha
+export const TechDocsAddons: React_2.ComponentType;
+
+// @public
+export type TechDocsEntityMetadata = Entity & {
+  locationMetadata?: {
+    type: string;
+    target: string;
+  };
+};
+
+// @public
+export type TechDocsMetadata = {
+  site_name: string;
+  site_description: string;
+};
+
+// @alpha (undocumented)
+export const TechDocsReaderPageContext: Context<
```

Comment:
Do not export Contexts directly unless you want to be in a world of versioning pain 😅  Better to exports provider and consumer components/hooks instead. By exporting this you pretty much freeze the API, ending up very limited in the changes you can do without breakage.

Reply:
Hum...  I think we ended up with this because the values can only be provided with access to TechDocs plugin-specific stuff (like the TechDocs API Ref).  ...Although it looks maybe possible to bring that API ref into this `-react` package and thus, bring the provider over too.

Would you recommend going down that route?

Reply:
Hmm, if you want to keep the provider where it is you can use the fact that the versioned context deduplicates things for you by the ID. We do that with the `core-app-api`/`core-plugin-api` split for example, with the producing APIs generally existing in the app and consumers in the plugin API. It looks like that would make sense in this case as well, as you really just need to make the hook available to consumers right? You need to duplicate some types, but that's generally fine as long as you keep things in sync :grin: